bindings/redis/redis.go

```diff
@@ -42,14 +45,25 @@ func NewRedis(logger logger.Logger) bindings.OutputBinding {
 
 // Init performs metadata parsing and connection creation.
 func (r *Redis) Init(meta bindings.Metadata) (err error) {
-	r.client, r.clientSettings, err = rediscomponent.ParseClientFromProperties(meta.Properties, nil)
+	if rediscomponent.IsLegacyRedisVersion(meta.Properties) {
+		r.legacyRedis = true
+		r.clientv8, r.clientSettings, err = rediscomponent.ParseClientv8FromProperties(meta.Properties, nil)
+	} else {
+		r.legacyRedis = false
+		r.clientv9, r.clientSettings, err = rediscomponent.ParseClientv9FromProperties(meta.Properties, nil)
+		r.logger.Warnf("Redis version 7 and above uses a Beta SDK at this time. Please use caution.")
+	}
 	if err != nil {
 		return err
 	}
 
 	r.ctx, r.cancel = context.WithCancel(context.Background())
 
-	_, err = r.client.Ping(r.ctx).Result()
+	if r.legacyRedis {
+		_, err = r.clientv8.Ping(r.ctx).Result()
```

Comment:
Is there a way to use Go lang to avoid these it checks? Could we declare an interface containing the methods we use here and declare a reference to that? Then, the only difference would be which instanced is referenced but they would both implement the same interface.